doc/example/rationale.cpp

```diff
@@ -9,6 +9,19 @@
 #include <turtle/mock.hpp>
 #include "calculator.hpp"
 #include "mock_view.hpp"
+#include <boost/test/unit_test.hpp>
+#include <iostream>
+#include <limits>
+#include <stdexcept>
+
+// Dummy to detect if the assertion unexpectedly succeeded to test what is explained
+#undef BOOST_CHECK_THROW
+#define BOOST_CHECK_THROW(expr, exc) \
```

Comment:
Maybe we could collect all this workarounds to be able to test the examples/patterns in one location and use them consistently from everywhere?

Reply:
Not sure. Each is only used once so I wanted to keep them very local as they are mostly hacks to be able to test the tests. Splitting them into other files might make it even harder to understand.